src/components/text-area/text-area.tsx

```diff
@@ -0,0 +1,476 @@
+import {
+  Component,
+  h,
+  Prop,
+  Element,
+  Event,
+  EventEmitter,
+  VNode,
+  Watch,
+  Method,
+  Host,
+  State
+} from "@stencil/core";
+import { connectForm, disconnectForm, FormComponent, HiddenFormInputSlot } from "../../utils/form";
+import { connectLabel, disconnectLabel, getLabelText, LabelableComponent } from "../../utils/label";
+import { slotChangeHasAssignedElement, toAriaBoolean } from "../../utils/dom";
+import { CSS, SLOTS, RESIZE_TIMEOUT } from "./resources";
+import {
+  connectLocalized,
+  disconnectLocalized,
+  LocalizedComponent,
+  NumberingSystem,
+  numberStringFormatter
+} from "../../utils/locale";
+import { createObserver } from "../../utils/observers";
+import {
+  componentLoaded,
+  LoadableComponent,
+  setComponentLoaded,
+  setUpLoadableComponent
+} from "../../utils/loadable";
+import {
+  connectMessages,
+  disconnectMessages,
+  setUpMessages,
+  T9nComponent,
+  updateMessages
+} from "../../utils/t9n";
+import { TextAreaMessages } from "./assets/text-area/t9n";
+import { throttle } from "lodash-es";
+import { InteractiveComponent, updateHostInteraction } from "../../utils/interactive";
+
+/**
+ * @slot - A slot for adding text.
+ * @slot footer-start - A slot for adding a leading footer.
+ * @slot footer-end - A slot for adding a trailing footer.
+ */
+
+@Component({
+  tag: "calcite-text-area",
+  styleUrl: "text-area.scss",
+  shadow: true,
+  assetsDirs: ["assets"]
+})
+export class TextArea
+  implements
+    FormComponent,
+    LabelableComponent,
+    LocalizedComponent,
+    LoadableComponent,
+    T9nComponent,
+    InteractiveComponent
+{
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteTextAreaElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When `true`, the component is focused on page load.
```

Comment:
Could this be the case if the component is added to the DOM after load? If so, perhaps mention initialization? 

> When `true`, the component is focused on initialization.

Reply:
I don't think if initialization captures the meaning. When the page is loaded with components  rendered  then the autofocus will kick in and applied to the component. Initialization could mean it is first attached to the DOM not essentially rendered. Thoughts?

Reply:
Ah okay, how about adding another line for context instead?:

```
When `true`, the component is focused on page load. Only one element can contain `autofocus`. If multiple elements have `autofocus`, the first element will receive focus.
```

Reply:
we are adding a reference to `mdn` website which provides the additional information. do we want to add it in here as well?

Reply:
Interesting.  Will add the additional context. FWIW, this needs to be updated for input components as well.

Reply:
I plan on supporting `@mdn` as a custom jsdoc tag so we can link to it in the API Ref when our prop is passed to a native attribute